components/kubernetes/kind.go

```diff
@@ -22,76 +24,81 @@ const (
 //go:embed kind-cluster.yaml
 var kindClusterConfig string
 
-// Install Kind on a Linux virtual machine
-func NewKindCluster(vm *vm.UnixVM, clusterName, arch, KubernetesVersion string) (*remote.Command, pulumi.StringOutput, error) {
-	runner := vm.GetRunner()
-	commonEnvironment := vm.GetCommonEnvironment()
-	packageManager := vm.GetPackageManager()
-	curlCommand, err := packageManager.Ensure("curl")
-	if err != nil {
-		return nil, pulumi.StringOutput{}, err
-	}
-	cmd, err := vm.GetLazyDocker().Install()
-	if err != nil {
-		return nil, pulumi.StringOutput{}, err
-	}
+// Install Kind on a Linux virtual machine.
+func NewKindCluster(env config.CommonEnvironment, vm *remote.Host, clusterName, kubeVersion string, opts ...pulumi.ResourceOption) (*Cluster, error) {
+	return components.NewComponent(env, clusterName, func(clusterComp *Cluster) error {
+		opts = utils.MergeOptions[pulumi.ResourceOption](opts, pulumi.Parent(clusterComp))
 
-	kindVersionConfig, err := getKindVersionConfig(KubernetesVersion)
-	if err != nil {
-		return nil, pulumi.StringOutput{}, err
-	}
+		runner := vm.OS.Runner()
+		commonEnvironment := env
+		packageManager := vm.OS.PackageManager()
+		curlCommand, err := packageManager.Ensure("curl", opts...)
+		if err != nil {
+			return err
+		}
 
-	kindInstall, err := runner.Command(
-		commonEnvironment.CommonNamer.ResourceName("kind-install"),
-		&command.Args{
-			Create: pulumi.String(fmt.Sprintf(`curl -Lo ./kind "https://kind.sigs.k8s.io/dl/%s/kind-linux-%s" && sudo install kind /usr/local/bin/kind`, kindVersionConfig.kindVersion, arch)),
-		},
-		utils.PulumiDependsOn(cmd, curlCommand),
-	)
-	if err != nil {
-		return nil, pulumi.StringOutput{}, err
-	}
+		_, dockerInstallCmd, err := docker.NewManager(env, vm, true, opts...)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You might want to return also the `installCompose` cmd, or a list of resources that `kindInstall` should depend on

Reply:
That's the beauty of components, it's not necessary. As they're all child of the `Cluster` component, external users only need to depend on `Cluster` and it will automatically wait on all children.

Reply:
What I meant is you were not returning the `docker-compose` cmd, but it is fixed now